### PR TITLE
Fixes 94: repo validation api

### PR DESCRIPTION
**Pull request by jlsherrill**

### CODE REVIEW

**Comment by jlsherrill** [comment]
https://issues.redhat.com/browse/HMSCONTENT-94

**Comment by jlsherrill** [comment]
example usage:
```
 curl  -v -X POST  -H "`./scripts/header.sh 1 1`"  http://localhost:8000/api/content_sources/v1/repositories/validate_parameters/   -H "Content-Type: application/json"    -d '[{"name": "foo", "url": "https://jlsherrill.fedorapeople.org/fake-repos/needed-errata/" }]'
```

If you want to test a timeout, you can run this in one terminal:
```
nc -lkp 4000
```

and then try to point a repo to it:
```
curl  -v -X POST  -H "`./scripts/header.sh 1 1`"  http://localhost:8000/api/content_sources/v1/repositories/validate_parameters/   -H "Content-Type: application/json"    -d '[{"name": "foo", "url": "https://localhost:4000/fake/" }]'
```

**Comment by mshriver** [comment]
[test]

**Review by Andrewgdewar — APPROVED** [review]
This is a "functional" ACK, I have this running smoothly with [my front-end branch](https://github.com/content-services/content-sources-frontend/pull/9) 
If someone with a more discerning eye would like to take a look at the code (which looks fine to me), please do. @rverdile  😉

**Comment by rverdile** [comment]
Sure I'll look

File: pkg/handler/repository_test.go [hunk 1]
```diff
@@ -666,6 +671,13 @@ func (suite *ReposSuite) TestPartialUpdate() {
 	assert.Equal(t, http.StatusOK, code)
 }
 
+//func (suite *ReposSuite) TestValidateParameters() {
+//	t := suite.T()
+//
+//	mockDao := MockRepositoryDao{}
+//	mockDao.On("ValidateParameters", mockOrgId, uuid, expected).Return(nil)
+//}
+
```
Comment (rverdile) [inline pkg/handler/repository_test.go:680]: Did you mean to come back to this?

File: pkg/dao/external_resource_test.go [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/mock"
+)
+
+type mockExternalResource struct {
+	mock.Mock
+}
+
+func (erd *mockExternalResource) ValidRepoMD(url string) (int, error) {
```
Comment (rverdile) [inline pkg/dao/external_resource_test.go:17]: Nitpick, but I think all the instances of "RepoMD" should be "Repomd"

File: pkg/api/repository_parameters.go [hunk 3]
```diff
@@ -7,3 +7,28 @@ type RepositoryParameterResponse struct {
...
+
+//Validation request/response
+type RepositoryValidationRequest struct {
+	Name *string `json:"name"`
+	URL  *string `json:"url"` // URL of the remote yum repository
+}
+
+type RepositoryValidationResponse struct {
+	Name GenericAttributeValidationResponse `json:"name"`
+	URL  UrlValidationResponse              `json:"url"`
```
Comment (rverdile) [inline pkg/api/repository_parameters.go:19]: Maybe these fields should get comments/documented. Something like "Validation response for repository name". I was a little confused at first. I feel like it's only intuitive from the naming in hindsight.

File: pkg/dao/repository_test.go [hunk 4]
```diff
@@ -661,3 +661,111 @@ func (suite *RepositorySuite) TestDeleteNotFound() {
...
+}
+
+func (e MockTimeoutError) Error() string {
+	return e.Message
+}
+
+func (suite *RepositorySuite) TestValidateParameters() {
+	t := suite.T()
+	org_id := "900023"
+	err := seeds.SeedRepositoryConfigurations(suite.tx /* &repo,*/, 1, seeds.SeedOptions{OrgID: org_id})
```
Comment (rverdile) [inline pkg/dao/repository_test.go:677]:
```suggestion
	err := seeds.SeedRepositoryConfigurations(suite.tx, 1, seeds.SeedOptions{OrgID: org_id})
```

File: pkg/dao/external_resource_test.go (hunk 2 above)
Comment (jlsherrill) [reply pkg/dao/external_resource_test.go:17]: so "MD" is short for "Metadata".  So i'd think it'd be similar to URL, which we've capitalized, but maybe Md would make more sense?  "RepoMd"

File: pkg/handler/repository_test.go (hunk 1 above)
Comment (jlsherrill) [reply pkg/handler/repository_test.go:680]: man i'm going senile in my old age

Comment (jlsherrill) [reply pkg/handler/repository_test.go:680]: ah yes, i wrote the  handler test over in repository_parameter_test, forgot to remove this

File: pkg/dao/external_resource_test.go (hunk 2 above)
Comment (rverdile) [reply pkg/dao/external_resource_test.go:17]: When I google it I see either "repomd" or "Repomd", which made me think it should be all lowercase. But if you want capital letter I think MD is fine.

**Review by rverdile — APPROVED** [review]
nice! code looks good

you can decide if you want RepoMD or RepoMd or Repomd :). I just wanted to mention it, but it's not a big deal.

File: pkg/dao/external_resource_test.go (hunk 2 above)
Comment (swadeley) [reply pkg/dao/external_resource_test.go:17]: -1 to  "RepoMd", so I suggest just leave it as is.

**Review by swadeley — APPROVED** [review]
ACK

**Pull request merged**